spotifython/playlist.py

```
@@ -1,228 +1,350 @@
-from user import User
-from track import Track
-from copy import deepcopy
-from typing import Union, List
+""" The Playlist class.
+
+This class represents a playlist as defined by Spotify.
+"""
 
-# objects created in this constructor
 class Playlist:
-    """
-    A Playlist object.
+    """ A Playlist object. """
 
-    Raises TypeError for incorrect types and ValueError for inappropriate
-    values.
-    """
+    # TODO store only static fields
+    def __init__(self, session, info):
+        """ Playlist constructor that should never be called directly.
 
-    # GET https://api.spotify.com/v1/playlists/{playlist_id}
-    def __init__(self, playlist_info: dict):
-        """
-        Playlist constructor that should never be called directly.
+        Args:
+            session: The Spotifython session object used to create this object.
+            info: The dict with playlist data produced from a Spotify API call.
         """
-        self._raw = deepcopy(playlist_info)
-        self._owner = User(playlist_info.get('owner', {}))
+        self._raw = deepcopy(info)
+        self._session = session
+        self._owner = User(session, info.get('owner', {}))
```

User raises an exception if `id` not in info. This call should probably raise its own exception instead of passing  in an empty dict